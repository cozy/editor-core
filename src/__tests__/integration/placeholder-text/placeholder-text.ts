import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';
import {
  editable,
  getDocFromElement,
  fullpage,
  insertBlockMenuItem,
} from '../_helpers';
import placeholderAdf from './__fixtures__/placeholder.adf.json';

const placeholderInputSelector = 'input[placeholder="Add placeholder text"]';
const placeholderNodeSelector = '[data-placeholder]';

BrowserTestCase(
  'placeholder text: can insert placeholder',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowTemplatePlaceholders: {
        allowInserting: true,
      },
    });

    await page.click(fullpage.placeholder);
    await insertBlockMenuItem(page, 'Placeholder text');

    await page.type(placeholderInputSelector, 'Fill this in...');
    await page.keys(['Enter']);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'placeholder text: can type inside placeholder and replace with text',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowTemplatePlaceholders: {
        allowInserting: true,
      },
      defaultValue: placeholderAdf,
    });

    await page.click(placeholderNodeSelector);
    await page.type(fullpage.placeholder, 'This is my text');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
