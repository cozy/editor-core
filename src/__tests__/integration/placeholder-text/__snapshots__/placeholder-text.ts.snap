// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`placeholder text: can insert placeholder 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "text": "Fill this in...",
          },
          "type": "placeholder",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`placeholder text: can type inside placeholder and replace with text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is my text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
